#165273590 User can highlight any text in an article and comment

nadralia opened the pull request:
### What does this PR do?
 User can highlight any text in an article and comment

### How to manually test these features

- Clone the Repo
- Make Migrations
- Run the server locally with python manage.py runserver
- Open postman and follow the instructions below :
- Create a user and login to the system using `http://localhost:8000/api/users/ `and `http://localhost:8000/api/users/login/ `respectively with POST methods
- Creating an article using POST method on 
   `http://localhost:8000/api/articles/`
- Create comment Using POST method with this URL `http://localhost:8000/api/<articleslug>/comments/` 

Comment body 
```
 { 
     "comment"    :{
	"body":"my first comments of today",
	"start_index":5,
	"end_index":13
       }
}
```
### Description of tasks to be completed
As a user, I should be able to highlight any text in an article(created by me or not) and comment on it.

### Relevant pivotal tracker story

[#165273590](https://www.pivotaltracker.com/story/show/165273590)

### Screenshots

Highlight and comment on text
<img width="1440" alt="Screenshot 2019-05-07 at 19 27 01 2" src="https://user-images.githubusercontent.com/38062858/57316491-9c6e7700-70fe-11e9-8627-2e77c0b07c2b.png">

bisonlou commented on authors/apps/comments/views.py:
```diff
@@ -5,37 +5,59 @@
+        start_index = comment.get('start_index', 0)
+        if "start_index" in comment and "end_index" in comment:
+            check = validationData(end_index,start_index,article)
+            if check == True:
```
check is not a very meaningful variable name. You could try [is_data_valid]

nadralia replied:
Okay let me change that

kalsmic replied:
> > well done, @nadralia, index validation on posting works fine but on updating the comment with an invalid invalid index in the request body, it also updates the index for the comment in the table
> > <img alt="Screenshot 2019-05-08 at 12 06 06" width="1161" src="https://user-images.githubusercontent.com/24764476/57363397-b30de000-7189-11e9-90eb-c33d9603dc22.png">
> 
> Let me see how I can override the update function so that I can I eliminate this issue. Thanks for pointing it out

Kudos, @nadralia, I love the way you have compared the new comment against the one existing in the database So that when the user provides an update to the comment that is similar to the current one, No changes are made to the comment. Algorithms +1 👍

kalsmic commented:
well done, @nadralia, index validation on posting works fine but on updating the comment with an invalid invalid index in the request body, it also updates the index for the comment in the table
<img width="1161" alt="Screenshot 2019-05-08 at 12 06 06" src="https://user-images.githubusercontent.com/24764476/57363397-b30de000-7189-11e9-90eb-c33d9603dc22.png">

nadralia commented:
> well done, @nadralia, index validation on posting works fine but on updating the comment with an invalid invalid index in the request body, it also updates the index for the comment in the table
> <img alt="Screenshot 2019-05-08 at 12 06 06" width="1161" src="https://user-images.githubusercontent.com/24764476/57363397-b30de000-7189-11e9-90eb-c33d9603dc22.png">

Let me see how I can override the update function so that I can I eliminate this issue. Thanks for pointing it out

llwasampijja reviewed (approved):
Nice work @nadralia!
Works as expected!

nadralia committed:
ft(highlight and comment):User can highlight any text in an article and comment

User can highlight text
User can comment on highlighted text
Test highlight and comment functionality

[Finishes #165273590]
